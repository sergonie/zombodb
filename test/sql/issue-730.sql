CREATE TABLE cats(name text);
INSERT INTO cats VALUES ('a'), ('b'), ('c');

CREATE TYPE cat_idx_type AS (name text);

CREATE OR REPLACE FUNCTION cat_idx_func(text) RETURNS cat_idx_type IMMUTABLE STRICT PARALLEL SAFE LANGUAGE sql AS $$
SELECT ROW($1)::cat_idx_type;
$$;

CREATE INDEX cats_idx ON cats USING zombodb (cat_idx_func(name)) WITH (url='http://localhost:9200/');

EXPLAIN
SELECT * FROM cats where cat_idx_func(name) ==> dsl.row_estimate(2,'cat');

DROP TABLE cats;
DROP TYPE cat_idx_type CASCADE;
