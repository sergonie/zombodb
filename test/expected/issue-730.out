CREATE TABLE cats(name text);
INSERT INTO cats VALUES ('a'), ('b'), ('c');
CREATE TYPE cat_idx_type AS (name text);
CREATE OR REPLACE FUNCTION cat_idx_func(text) RETURNS cat_idx_type IMMUTABLE STRICT PARALLEL SAFE LANGUAGE sql AS $$
SELECT ROW($1)::cat_idx_type;
$$;
CREATE INDEX cats_idx ON cats USING zombodb (cat_idx_func(name)) WITH (url='http://localhost:9200/');
EXPLAIN
SELECT * FROM cats where cat_idx_func(name) ==> dsl.row_estimate(2,'cat');
                                              QUERY PLAN                                              
------------------------------------------------------------------------------------------------------
 Index Scan using cats_idx on cats  (cost=0.00..0.03 rows=2 width=32)
   Index Cond: (name ==> '{"row_estimate":2,"query_dsl":{"query_string":{"query":"cat"}}}'::zdbquery)
(2 rows)

DROP TABLE cats;
DROP TYPE cat_idx_type CASCADE;
